test/features/test_organized_edge_detection.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#include <pcl/test/gtest.h>
+#include <pcl/point_cloud.h>
+#include <pcl/features/organized_edge_detection.h>
+#include <pcl/io/pcd_io.h>
+
+namespace
+{
+  class OrganizedPlaneDetectionTestFixture : public ::testing::Test
+  {
+   protected:
+     const int kSyntheticCloudInnerSquareEdgeLength = 124;
+     const int kSyntheticCloudOuterSquareEdgeLength = kSyntheticCloudInnerSquareEdgeLength * 2;
+
+     const int kExpectedOccludingEdgePoints = kSyntheticCloudInnerSquareEdgeLength * 4 - 8;  // Empirically determined
+     const int kExpectedOccludedEdgePoints = kSyntheticCloudInnerSquareEdgeLength * 4;       // Each edge pixed of inner square should generate an occluded point
+
+     const float kSyntheticCloudDisparity = .03f;
```

Comment:
There's no point in calling this disparity. It's just a depth difference the plane of both squares. Disparity in computer vision is a concept which relates to stereo imaging. This entire test is built with a single RGBD image so calling this disparity is misleading.